service/history/shard/controller_impl.go

```diff
@@ -353,10 +353,16 @@ func (c *ControllerImpl) acquireShards() {
 			c.contextTaggedLogger.Error("Unable to create history shard context", tag.Error(err), tag.OperationFailed, tag.ShardID(shardID))
 			return
 		}
+
 		// Wait up to 1s for the shard to acquire the rangeid lock.
 		// After 1s we will move on but the shard will continue trying in the background.
 		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
 		defer cancel()
+		// trust the AssertOwnership will handle shard ownership lost
+		_ = shard.AssertOwnership(ctx)
+
+		ctx, cancel = context.WithTimeout(context.Background(), shardIOTimeout)
```

Comment:
wait, now the timeouts are wrong.. it's using 1s for AssertOwnership and 5s for GetEngine. was that the intention?

what was wrong with the suggestion to do GetEngine and then if it succeeds, then do AssertOwnership?

Reply:
>what was wrong with the suggestion to do GetEngine and then if it succeeds, then do AssertOwnership?

1. the order does not matter
2. i already updated the logic

Reply:
>wait, now the timeouts are wrong.. it's using 1s for AssertOwnership and 5s for GetEngine. was that the intention?

i am using shardIOTimeout for shard assertion, this operation should be really fast, compared to engine start

Reply:
yeah, I made that comment on an intermediate commit when they were different. this looks good now.